### PR TITLE
core(total-byte-weight): count partially finished requests

### LINKED ISSUE #12664: Clarification about "total-byte-weight" audit
Hi there !

We are currently integrating lighthouse in our CI / CD process and running some benchmarks with it, thanks to lighthouse CLI.
On the same report, we have different values for total-byte-weight measure analyzing json file.

Let say that we are performing an audit on a website and we are using the total-byte-weight metric.
For some reasons, I have the total byte weight audit result saying me :

`total-byte-weight: {
id: "total-byte-weight",
title: "Avoids enormous network payloads",
description: "Large network payloads cost users real money and are highly correlated with long load times. [Learn more](https://web.dev/total-byte-weight/).",
score: 1,
scoreDisplayMode: "numeric",
numericValue: 714213,
numericUnit: "byte",
displayValue: "Total size was 697 KiB",`

But, when I look at the totalbyteweight property into audit.diagnostics.details.items, I get the value : 6795874

The "real" page total byte weight is, I think, 6,48MiB.
So, could you please help me understanding the difference with the 2 values ? Could it be related to the service worker on the website ?

You will find attached the json report file.

[strange-tbw.json.zip](https://github.com/GoogleChrome/lighthouse/files/6658122/strange-tbw.json.zip)

Thanks a lot for your help.

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Removes the total byte weight policy that ignored partially downloaded resources (like videos). IMO, this dramatically understates the weight of a page in some cases (see #12664).

The check was added [way back in 2017](https://github.com/GoogleChrome/lighthouse/pull/2023#discussion_r114909191) _by me_ as part of __adding visibility into inflight records__ 🤯 . This was also before we used our own network request class that initializes transferSize that prevents the `undefined` concern that prompted the original.

Nonetheless, I added a test to ensure spirit of the condition is still handled.

**Related Issues/PRs**
closes #12664

### CODE REVIEW

File: lighthouse-core/test/network-records-to-devtools-log.js [hunk 1]
```diff
@@ -115,7 +115,8 @@ function getResponseReceivedEvent(networkRecord, index) {
         connectionId: networkRecord.connectionId || 140,
         fromDiskCache: networkRecord.fromDiskCache || false,
         fromServiceWorker: networkRecord.fetchedViaServiceWorker || false,
-        encodedDataLength: networkRecord.transferSize || 0,
+        encodedDataLength: typeof networkRecord.transferSize === 'undefined' ?
```
Comment (connorjclark) [inline lighthouse-core/test/network-records-to-devtools-log.js:118]: I didn't understand the note about `undefined`. Why does this need to change?

Don't need to do typeof here, just `networkRecord.transferSize === undefined` works. Or if the data type is somehow uncertain, flip the logic to `typeof networkRecord.transferSize === 'number' ? networkRecord.transferSize : 0`.

Comment (patrickhulce) [reply lighthouse-core/test/network-records-to-devtools-log.js:118]: > Don't need to do typeof here

ah yes, old habits :)

> Or if the data type is somehow uncertain, flip the logic to

The spirit of the original I assumed was `?? 0`, i.e. use `0` as a default if some value is not explicitly set. That seems better than ignoring all values that aren't numbers.

> I didn't understand the note about undefined. Why does this need to change?

If we are ever to test unexpected values coming from the protocol, we need a mechanism to mock those values in the devtools log. The affected code was attempting to handle such unexpected values.

**Review by jjvrmorales367 — APPROVED** [review]
i'd try this out and see if anything changes

**Review by jjvrmorales367 — APPROVED** [review]
I'm still yet to try this but it all goods

File: lighthouse-core/test/audits/byte-efficiency/total-byte-weight-test.js [hunk 2]
```diff
@@ -90,4 +90,25 @@ describe('Total byte weight audit', () => {
...
+      {url: 'https://example.com/file.js', transferSize: 10_000},
+      {url: 'data:image/jpeg;base64,', protocol: 'data', transferSize: 100_000},
+      {url: 'blob:1234', protocol: 'blob', transferSize: 100_000},
+    ]);
+
+    const result = await TotalByteWeight.audit(artifacts, {options, computedCache: new Map()});
+    expect(result.numericValue).toEqual(10_000);
+  });
+
+  it('does not throw on requests with NaN transferSize', async () => {
```
Comment (adamraine) [inline lighthouse-core/test/audits/byte-efficiency/total-byte-weight-test.js:105]: nit: I was expecting something like
```suggestion
  it('ignores requests with falsy transfer size', async () => {
```

**Review by adamraine — APPROVED** [review]

**Pull request merged**